### PR TITLE
Resolve some extra warnings

### PR DESCRIPTION
**Pull request by seelabs**
This is the new PR for https://github.com/ripple/rippled/pull/2504 and https://github.com/ripple/rippled/pull/2503

### CODE REVIEW

**Comment by seelabs** [comment]
LGTM: :+1:

File: src/beast/include/beast/websocket/stream.hpp [hunk 1]
```diff
@@ -617,7 +617,7 @@ class stream
             BOOST_THROW_EXCEPTION(std::invalid_argument{
                 "write buffer size underflow"});
         wr_buf_size_ = n;
-    };
```
Comment (mellery451) [inline src/beast/include/beast/websocket/stream.hpp]: this is changing submodule'd code...and will be removed soon anyhow ?

File: src/nudb/include/nudb/progress.hpp [hunk 2]
```diff
@@ -24,7 +24,7 @@ no_progress
     void
     operator()(std::uint64_t, std::uint64_t) const noexcept
     {
-    };
```
Comment (mellery451) [inline src/nudb/include/nudb/progress.hpp]: also a submodule...

**Comment by JoeLoser** [comment]
Sorry for updating third-party code with that small change. I'll avoid that in the future.

**Comment by mellery451** [comment]
@JoeLoser -- not a problem...it's sadly not completely obvious from the repo layout that these are in-fact submodules/NIH sources

**Comment by JoeLoser** [comment]
Probably wouldn't hurt to just replace the Git submodule paths with a top-level `vendor` or `third-party` folder? It'll just be for `Rocksdb` and `Nudb` soon anyway once the dev branch is updated to use Beast from Boost.

**Comment by MarkusTeufelberger** [comment]
Vendored code bases are all folders in https://github.com/ripple/rippled/tree/develop/src except "test" and "ripple".

**Comment by JoeLoser** [comment]
Ah, I see. I was just naively looking at the Git submodules. Makes sense now. Thanks!

**Comment by nbougalis** [comment]
Merged as c1c332f0b009158b5dd791acd3db5de2901b20e1 and 04f13888608535c4d0fe47cffed53c33b0a33b69.

**Pull request closed without merging**